receiver/otlpreceiver/otlp.go

```diff
@@ -186,8 +187,18 @@ func (r *otlpReceiver) Start(ctx context.Context, host component.Host) error {
 func (r *otlpReceiver) Shutdown(ctx context.Context) error {
 	var err error
 
+	if r.cfg != nil {
+		if r.cfg.HTTP != nil && r.cfg.HTTP.TLSSetting != nil {
+			err = r.cfg.HTTP.TLSSetting.Shutdown()
+		}
+
+		if r.cfg.GRPC != nil && r.cfg.GRPC.TLSSetting != nil {
+			err = multierr.Append(err, r.cfg.GRPC.TLSSetting.Shutdown())
+		}
+	}
+
 	if r.serverHTTP != nil {
-		err = r.serverHTTP.Shutdown(ctx)
+		err = multierr.Append(err, r.serverHTTP.Shutdown(ctx))
```

Comment:
I'd prefer `errors.Join` instead of `multierr`.

Reply:
Agreed, updated.

Reply:
If you don't want the resulting error to be nested I believe you need to keep a slice of errors and do the `errors.Join` once at the end.  I'm not sure here if that matters or not.

Reply:
Thanks for catching that, I've updated.